editor/src/main/java/com/kotcrab/vis/editor/ui/scene/entityproperties/BasicEntityPropertiesTable.java

```diff
@@ -27,374 +35,729 @@
 import com.kotcrab.vis.editor.util.value.FloatProxyValue;
 import com.kotcrab.vis.editor.util.vis.EntityUtils;
 import com.kotcrab.vis.runtime.util.ImmutableArray;
+import com.kotcrab.vis.ui.util.CursorManager;
 import com.kotcrab.vis.ui.util.TableUtils;
 import com.kotcrab.vis.ui.util.value.VisValue;
-import com.kotcrab.vis.ui.widget.Tooltip;
-import com.kotcrab.vis.ui.widget.VisImage;
-import com.kotcrab.vis.ui.widget.VisLabel;
-import com.kotcrab.vis.ui.widget.VisTable;
+import com.kotcrab.vis.ui.widget.*;
 import com.kotcrab.vis.ui.widget.color.ColorPicker;
 import com.kotcrab.vis.ui.widget.color.ColorPickerAdapter;
 import com.kotcrab.vis.ui.widget.color.ColorPickerListener;
+import org.lwjgl.input.Keyboard;
 
 import java.util.EnumSet;
 
 import static com.kotcrab.vis.editor.ui.scene.entityproperties.EntityProperties.*;
 
-/** @author Kotcrab */
+/**
+ * @author Kotcrab
+ */
 public class BasicEntityPropertiesTable extends VisTable {
-	private final EntityProperties properties;
-	private final ColorPicker picker;
-
-	private ColorPickerListener pickerListener;
-	private TintImage tint;
-
-	private VisTable idTable;
-
-	private VisTable positionTable;
-	private VisTable scaleTable;
-	private VisTable originTable;
-	private VisTable rotationTable;
-	private VisTable tintTable;
-	private VisTable flipTable;
-
-	private IndeterminateTextField idField;
-	private NumberInputField xField;
-	private NumberInputField yField;
-	private NumberInputField xScaleField;
-	private NumberInputField yScaleField;
-	private NumberInputField xOriginField;
-	private NumberInputField yOriginField;
-	private NumberInputField rotationField;
-	private IndeterminateCheckbox xFlipCheck;
-	private IndeterminateCheckbox yFlipCheck;
-
-	private VisImage positionLock = createLockImage();
-	private VisImage originLock = createLockImage();
-	private VisImage scaleLock = createLockImage();
-	private VisImage rotationLock = createLockImage();
-
-	private EnumSet<LockableField> lockedFields = EnumSet.noneOf(LockableField.class);
-
-	public BasicEntityPropertiesTable (EntityProperties properties, ColorPicker picker) {
-		this.properties = properties;
-		this.picker = picker;
-
-		createIdTable();
-		createPositionTable();
-		createScaleTable();
-		createOriginTable();
-		createRotationTintTable();
-		createFlipTable();
-
-		pickerListener = new ColorPickerAdapter() {
+    private final EntityProperties properties;
```

Comment:
@soywiz  somehow `.editorconfig` didn't work for this file. Spaces instead of tabs!